### DISCUSSION TITLE
Modal: Can't be closed by clicking above

### DISCUSSION THREAD

**Discussion by martinkrulltott** [dhis2/notes#232] [Q&A] [answered]
Clicking directly above a modal doesn't trigger it to be closed.

As can be seen in this example: https://codesandbox.io/s/ui5-sandbox-j8fhs?file=/src/App.js

The `centered-inner-content` element seems to be the cause of this.

![image](https://user-images.githubusercontent.com/12590483/108480319-19002480-7297-11eb-8f76-b4a2382802d2.png)

**Comment by Mohammer5** [comment]
> This isn't really a Q/A issue, right?
> It's a bug report, you could just create an issue in Jira.

https://github.com/dhis2/notes/discussions/234#discussioncomment-383813

**Reply by martinkrulltott** [reply]
> IDK, maybe its per design? Or out of scope? Or something that needs to be discussed first? But if you want me to create a ticket for it then I will.

https://github.com/dhis2/notes/discussions/234#discussioncomment-384046

**Reply by martinkrulltott** [reply]
https://jira.dhis2.org/browse/LIBS-127

**Reply by Birkbjo** [reply]
Agree that this should be a bug-report. I don't know the implementation details as to why the inner-div is there (seems to be to prevent backdrop-clicks?). @HendrikThePendric, any ideas on how to fix this? A potential solution I played around with was to remove the inner-div completely, but that needed some changes on how to handle backdrop-clicks.

**Reply by martinkrulltott** [reply]
Maybe even just changing `pointer-events: all;` to `pointer-events: none;` would be enough?

**Reply by Birkbjo** [reply]
That would result in `onClose` be called when clicked on the modal as well. Seems like that is the sole purpose of the inner-div - to prevent that from happening,

**Reply by martinkrulltott** [reply]
yes, that's right, the first child (`aside`) would need the `pointer-events: all;` then to prevent the closing when the modal content is clicked.

**Comment by varl** [comment, marked as answer]
> https://jira.dhis2.org/browse/LIBS-127

Thanks for reporting this issue, @martinkrulltott. I'm copying it out to a "new answer" so we can mark it as the answer. It will be fixed.

**Comment by KaiVandivier** [comment]
Fixed in UI v6.5.1 👍
